File: problems/c/countingclauses/countingclauses.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    m, n = [int(val) for val in input().split()]

    for _ in range(m):
        _ = input()

    if m >= 8:
        print("satisfactory")
    else:
        print("unsatisfactory")



if __name__ == '__main__':
    main()

File: problems/c/countingclauses/countingclauses.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>

using namespace std;

int main() {
    int m, n;
    int clause;

    cin >> m >> n;

    for (int i=0; i<m; i++) {
        cin >> clause;
    }

    if (m>=8) {
        cout << "satisfactory" << endl;
    }
    else {
        cout << "unsatisfactory" << endl;
    }
}